cmd/tuf/server/main.go

```diff
@@ -140,26 +149,112 @@ func main() {
 	// worries here.
 	var compressed bytes.Buffer
 	if err := repo.CompressFS(os.DirFS(dir), &compressed, map[string]bool{"keys": true, "staged": true}); err != nil {
-		logging.FromContext(ctx).Fatalf("Failed to compress the repo: %v", err)
+		return fmt.Errorf("failed to compress the repo: %v", err)
 	}
 	data["repository"] = compressed.Bytes()
 
 	if !*noK8s {
 		nsSecret := clientset.CoreV1().Secrets(ns)
-		if err := secret.ReconcileSecret(ctx, *secretName, ns, data, nsSecret); err != nil {
-			logging.FromContext(ctx).Panicf("Failed to reconcile secret %s/%s: %v", ns, *secretName, err)
+		if err := secret.ReconcileSecret(ctx, repoSecretName, ns, data, nsSecret); err != nil {
+			return fmt.Errorf("failed to reconcile secret %s/%s: %v", ns, repoSecretName, err)
+		}
+
+		// If we should also store created keys in a secret, read all their files and save them in the secret
+		if keysSecretName != "" {
+			keyFiles, err := os.ReadDir(filepath.Join(dir, "keys"))
```

Comment:
Didn't we pack the `keys` directory into `data["repository"]` already, along with the rest of `dir`?

Reply:
We didn't. If you look at how this is done:

```
if err := repo.CompressFS(os.DirFS(dir), &compressed, map[string]bool{"keys": true, "staged": true}); err != nil
```

and how the `CompressFS` function works, it will actually intentionally skip the `keys` and `staged` directories because of how it is called.